File: 0234-palindrome-linked-list/0234-palindrome-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        string check="";
        while(head){
            check+=to_string(head->val);
            head=head->next;
        }
        int i=-1,j=check.size();
        while(i++<j--)if(check[i]!=check[j])return false;
        return true;
    }
};

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        StringBuilder check = new StringBuilder();
        while (head != null) {
            check.append(head.val);
            head = head.next;
        }
        int i = -1, j = check.length();
        while (++i < --j) {
            if (check.charAt(i) != check.charAt(j)) {
                return false;
            }
        }
        return true;
    }
}